[ios] Update Quick and Nimble dependencies

wschurman opened the pull request:
# Why

One feature I'm looking to use in nimble is assertion expectations: https://github.com/Quick/Nimble#swift-assertions

Support for these running on AMD was added in a more recent version.

# How

Upgrade deps.

# Test Plan

Wait for full test run.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

wschurman committed:
[ios] Update Quick and Nimble depdencies

tsapeta commented:
Afaik there were some breaking changes in one of the previous Quick versions, so this sounds like a bigger task 😉

wschurman commented:
Expo Go build is failing from base PR. Will fix there. This is ready to review though since it only touches unit tests.

ide commented on packages/expo-dev-launcher/ios/Tests/EXDevLauncherPendingDeepLinkRegistryTest.swift:
```diff
@@ -44,7 +44,7 @@ class EXDevLauncherPendingDeepLinkRegistryTest: QuickSpec {
       expect(registry.pendingDeepLink).to(beNil())
       expect(listener.lastDeepLink?.absoluteString).to(equal("http://localhost:1234"))
-      expect(listener.lastDeepLink).to(be(consumedURL))
+      expect((listener.lastDeepLink! as NSURL) === (consumedURL! as NSURL)).to(beTrue())
```
Why this change? Having the assertion library do the equality check produces a clearer error message

ide commented on packages/expo-modules-core/ios/Tests/ExceptionsSpec.swift:
```diff
@@ -73,7 +73,7 @@ final class ExceptionsSpec: ExpoSpec {
       let c = TestException().causedBy(b)
 
-      expect(c.rootCause) === a
+      expect((c.rootCause as! TestException) === a).to(beTrue())
```
```suggestion
      expect((c.rootCause as! TestException)).to(beEqual(a))
```
Something like this possible? (Suggestion is not real code)

ide commented on packages/expo-modules-core/ios/Tests/ExpoModulesSpec.swift:
```diff
@@ -61,7 +67,7 @@ class ExpoModulesSpec: ExpoSpec {
 
         dict.forEach { (key: String, value: Any) in
-          expect(value) === dict[key]!
+          expect((value as! NSObject) === dict[key]! as! NSObject).to(beTrue())
```
Same comment about having the assertion library do the equality check

wschurman replied:
done

ide commented on packages/expo-modules-core/ios/Tests/FunctionSpec.swift:
```diff
@@ -103,7 +103,8 @@ class FunctionSpec: ExpoSpec {
               expect(value).notTo(beNil())
               expect(value).to(beAKindOf(String.self))
-              expect(value).to(be(dict["property"]))
+//              expect(value).to(equal(dict["property"]))
```
Mean to revisit this?

ide commented on packages/expo-updates/ios/Tests/UpdateSpec.swift:
```diff
@@ -7,13 +7,13 @@ import ExpoModulesTestCore
+  override class func spec() {
+    let config = try! UpdatesConfig.config(fromDictionary: [
+      UpdatesConfig.EXUpdatesConfigRuntimeVersionKey: "1",
+      UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://exp.host/@test/test"
```
```suggestion
      UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://u.expo.dev/00000000-0000-0000-0000-000000000000"
```

Minor but can replace classic url with EAS

ide commented on packages/expo-updates/ios/Tests/NewUpdateSpec.swift:
```diff
@@ -7,13 +7,13 @@ import ExpoModulesTestCore
-  let database = UpdatesDatabase()
+  override class func spec() {
+    let config = try! UpdatesConfig.config(fromDictionary: [
+      UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://exp.host/@test/test",
```
```suggestion
      UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://u.expo.dev/00000000-0000-0000-0000-000000000000",
```

ide commented on packages/expo-updates/ios/Tests/LegacyUpdateSpec.swift:
```diff
@@ -7,17 +7,17 @@ import ExpoModulesTestCore
 import EXManifests
 
 class LegacyUpdateSpec : ExpoSpec {
```
Unrelated to this PR but do we need LegacyUpdate tests? Holding on to them till SDK 51's Expo Go drops 49?

wschurman replied:
Yep.

ide reviewed (approved):
Approving but left some smaller comments.

tsapeta commented:
> Expo Go build is failing from base PR. Will fix there. This is ready to review though since it only touches unit tests.

#25467 should fix this issue

wschurman commented on packages/expo-modules-core/ios/Tests/ExpoModulesSpec.swift:
```diff
@@ -4,8 +4,14 @@ import ExpoModulesTestCore
 @testable import ExpoModulesCore
 
+public extension Equatable {
+  static func equate(_ any0: some Any, _ any1: some Any) -> Bool {
```
Weird, seems like this was auto-added? I'll remove

wschurman commented on packages/expo-modules-core/ios/Tests/FunctionSpec.swift:
```diff
@@ -103,7 +103,8 @@ class FunctionSpec: ExpoSpec {
               expect(value).to(beAKindOf(String.self))
-              expect(value).to(be(dict["property"]))
+//              expect(value).to(equal(dict["property"]))
+              expect(value)
```
Yeah, sorry, this was a work-in-progress that I missed when making this ready for review. fixed

wschurman commented on apps/native-tests/ios/NativeTests.xcodeproj/project.pbxproj:
going to revert this file. no idea why it was generated this way but ld_classic seems wrong. will re-run `pod install` in a separate PR.

wschurman commented:
Cool, addressed comments and unit tests passed. E2E test failures look unrelated. Test suite timed out. Going to land after addressing self-comment to hopefully fix a few of the other failing tests suites.
